src/Views/Textile/Step.elm

```diff
@@ -680,128 +681,124 @@ viewStepImpacts selectedImpact { impacts, complementsImpacts } =
-                                , text "%"
+                    totalShares =
+                        inputs.materials
+                            |> List.map (.share >> Split.toFloat >> clamp 0 1)
+                            |> List.sum
+
+                    valid =
+                        round (totalShares * 100) == 100
+                 in
+                 li
+                    [ class "input-group "
+                    , classList [ ( "AddElementFormWrapper ps-3", length > 1 ) ]
+                    ]
+                    [ if length > 1 then
+                        span
+                            [ class "SharesTotal ext-end"
+                            , class "d-flex justify-content-between align-items-center gap-1"
+                            , classList
+                                [ ( "text-success feedback-valid", valid )
+                                , ( "text-danger feedback-invalid", not valid )
                                 ]
-
-                          else
-                            text ""
-                        , button
-                            [ class "AddElementButton btn btn-outline-primary flex-fill"
-                            , class "d-flex justify-content-center align-items-center gap-1 no-outline"
-                            , id "add-new-element"
-                            , onClick
-                                (setModal
-                                    (addMaterialModal Nothing
-                                        (AutocompleteSelector.init .shortName availableMaterials)
-                                    )
-                                )
-                            , disabled <| length >= Env.maxMaterials
                             ]
-                            [ Icon.plus
-                            , if length >= Env.maxMaterials then
-                                text "Nombre maximal de matières atteint"
+                            [ if valid then
+                                Icon.check
 
                               else
-                                text "Ajouter une matière"
+                                Icon.warning
+                            , round (totalShares * 100) |> String.fromInt |> text
+                            , text "%"
                             ]
+
+                      else
+                        text ""
+                    , button
+                        [ class "AddElementButton btn btn-outline-primary flex-fill"
+                        , class "d-flex justify-content-center align-items-center gap-1 no-outline"
+                        , id "add-new-element"
+                        , onClick
+                            (setModal
+                                (addMaterialModal Nothing
+                                    (AutocompleteSelector.init .shortName availableMaterials)
+                                )
+                            )
```

Comment:
nit: this can be written as

```elm
    , availableMaterials
        |> AutocompleteSelector.init .shortName
        |> addMaterialModal Nothing
        |> setModal
        |> onClick
```

I prefer this form but the one is the patch is fine too, so pick whatever you prefer